### PR TITLE
[50730] Show warning if you cannot add new users due to user limit

### PR DESCRIPTION
**Pull request by HDinger**
Currently we can invite new users inline via the users autocompleter in the share modal and on the members page. However, the instance might have a user limit that will be exceed by this.
This PR adds an information for users if they try to invite a new user and the limit will be exceeded. For the members page that already existed (but it was broken and fixed in this PR as well).

**Note:** There is only a warning. The users can still be invited, but they cannot login until the instance is upgraded.

### Todo
- [x] Show a warning if inviting new users would exceed the user limit (share modal)
- [x] Fix the mechanism for the warning on the members page
- [x] Add a test

https://community.openproject.org/projects/openproject/work_packages/50730/github

### CODE REVIEW

File: app/components/work_packages/share/invite_user_form_component.html.erb [hunk 1]
```diff
@@ -1,24 +1,41 @@
...
                                                                       form_arguments: { builder: form, name: "role_id" },
                                                                       data: { 'test-selector': 'op-share-wp-invite-role' }))
           end
 
-          new_user_row.with_column do
+          invite_form.with_area('submit') do
             render(Primer::Beta::Button.new(scheme: :primary, type: :submit)) { I18n.t('work_package.sharing.share') }
           end
+
+          unless OpenProject::Enterprise.user_limit.nil?
```
Comment (aaron-contreras) [inline app/components/work_packages/share/invite_user_form_component.html.erb]: Double negative. How about:
```ruby
if OpenProject::Enterprise.user_limit.present?
```

File: config/locales/en.yml [hunk 2]
```diff
@@ -3265,6 +3265,10 @@ en:
       share: "Share"
       text_empty_state_description: "The work package has not been shared with anyone yet."
       text_empty_state_header: "No shared users"
+      text_user_limit_reached:
+        zero: "This instance has reached its user limit, you can’t share with new users. Please contact your administrator to add more members."
+        one: "Only 1 member left to reach the user limit of the instance. Please contact your administrator."
+        other: "Only %{count} members left to reach the user limit of the instance. Please contact your administrator."
```
Comment (aaron-contreras) [inline config/locales/en.yml]: Unless you already talked about this, it'd be good to check with product about this copyright. Based on the PR description note, users can still invite (but invited users can't sign in). However, the `zero` case here would indicate to me I can't

File: app/components/work_packages/share/invite_user_form_component.html.erb (hunk 1 above)
Comment (HDinger) [reply app/components/work_packages/share/invite_user_form_component.html.erb]: 👍

File: config/locales/en.yml (hunk 2 above)
Comment (HDinger) [reply config/locales/en.yml]: I doubled checked with product and we adjusted the texts. 👍

**Review by aaron-contreras — APPROVED** [review]
💣!

**Pull request merged**